### PR TITLE
PLANNER-2316: Task assigning service scaffold and solution loading

### PR DESCRIPTION
**Pull request by wmedvede**
Many thanks for submitting your Pull Request :heart:! 

Please make sure that your PR meets the following requirements:

- [x] You have read the [contributors guide](CONTRIBUTING.md)
- [x] Your code is properly formatted according to [this configuration](https://github.com/kiegroup/droolsjbpm-build-bootstrap/tree/master/ide-configuration)
- [x] Pull Request title is properly formatted: `KOGITO-XYZ Subject`
- [x] Pull Request title contains the target branch if not targeting master: `[0.9.x] KOGITO-XYZ Subject`
- [x] Pull Request contains link to the JIRA issue
- [x] Pull Request contains link to any dependent or related Pull Request
- [x] Pull Request contains description of the issue
- [x] Pull Request does not include fixes for issues other than the main ticket

### CODE REVIEW

**Comment by wmedvede** [comment]
Hi @rsynek would you mind review this PR?

Thanks

File: kogito-bom/pom.xml [hunk 1]
```diff
@@ -1483,6 +1483,41 @@
         <classifier>sources</classifier>
       </dependency>
 
+      <!-- Kogito task assigning -->
```
Comment (rsynek) [inline kogito-bom/pom.xml:1486]: Usually the BOM declares also `<classifier>javadoc</classifier>` and `<classifier>test-jar</classifier>` dependencies to have the complete variety of artifacts.

Comment (wmedvede) [reply kogito-bom/pom.xml:1486]: I know, In fact I was about to add them, however I have seen that many modules are not declaring them at all, and in general I'd say that there aren't a well defined pattern in that sense. Considering also that this is a brand new thing, which might be subject to eventual refactorings, etc, I preferred  to add the minimal set to make this work.

But not a big deal to declare them

Comment (wmedvede) [reply kogito-bom/pom.xml:1486]: added

Comment (rsynek) [reply kogito-bom/pom.xml:1486]: Thanks, I noticed they are missing for many other artifacts in this BOM. The OptaPlanner BOM declares them more often, but even there it's not the case for every single dependency.

**Review by rsynek — APPROVED** [review]

**Comment by wmedvede** [comment]
damn, looks like I don't have merge grants.
@rsynek would you mind merge this if you have please?
thanks

**Comment by wmedvede** [comment]
Non related failure

  2350K .......... .......... .......... .......... ..........  182M

  2400K .......... .......... .......... .......... ..........  244M

  2450K .......... .......... .......... .......... .......... 13.7M

Body did not finish within grace period; terminating with extreme prejudice

Error when executing always post condition:

org.jenkinsci.plugins.workflow.steps.FlowInterruptedException

	at org.jenkinsci.plugins.workflow.steps.TimeoutStepExecution.cancel(TimeoutStepExecution.java:160)

	at org.jenkinsci.plugins.workflow.steps.TimeoutStepExecution.access$100(TimeoutStepExecution.java:38)

	at org.jenkinsci.plugins.workflow.steps.TimeoutStepExecution$1.run(TimeoutStepExecution.java:134)

	at jenkins.security.ImpersonatingScheduledExecutorService$1.run(ImpersonatingScheduledExecutorService.java:58)

	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)

	at java.util.concurrent.FutureTask.run(FutureTask.java:266)

	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:180)

	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:293)

	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)

	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)

	at java.lang.Thread.run(Thread.java:748)

**Comment by wmedvede** [comment]
Jenkins retest this please

**Pull request merged**